### DISCUSSION TITLE
SOLD2 - decoupling heatmap extraction and line segment detection

### DISCUSSION THREAD

**Discussion by ogencoglu** [kornia/kornia#2545] [Questions] [answered]
SOLD2 first extracts junction and line heatmaps (through a neural network) and then assigns line segments as a separate step. Is there a way to decouple these steps and perform them separately by calling different functions?

My use case would benefit a lot from being able to modify the line and junction heatmaps and then pass them to the line segment extractor.

**Comment by ducha-aiki** [comment, marked as answer]
@ogencoglu 
You can use 

```python3
sold2 = KF.SOLD2(pretrained=True, config=None)
stuff = sold2.model(img)
```

Check https://github.com/kornia/kornia/blob/a7fa3320c5a060d4bbf200c2d008fcc44c814624/kornia/feature/sold2/sold2.py#L116
